### PR TITLE
Added controllers, models, tests, repository

### PR DESCRIPTION
**Pull request by martinez-guillaume**
added controllers, models, tests, repository and CRUD operations on the Student in the file src/main/java/com/tajweed/backend/BackendApplication.java

### CODE REVIEW

File: src/main/java/com/tajweed/backend/BackendApplication.java [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+<<<<<<< Updated upstream
 import com.mongodb.client.MongoClient;
 import com.mongodb.client.internal.MongoClientImpl;
 import com.tajweed.backend.dao.StudentRepository;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+=======
+>>>>>>> Stashed changes
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/BackendApplication.java:13]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/BackendApplication.java [hunk 2]
```diff
@@ -1,12 +1,16 @@
 package com.tajweed.backend;
 
+import org.bson.types.ObjectId;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+<<<<<<< Updated upstream
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/BackendApplication.java:6]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/BackendApplication.java [hunk 3]
```diff
@@ -15,24 +19,80 @@
...
+	@Autowired
+	private StudentRepository StudentRepository;
+
+
+
 	public static void main(String[] args) {
 		SpringApplication.run(BackendApplication.class, args);
 	}
 
+<<<<<<< Updated upstream
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/BackendApplication.java:53]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/BackendApplication.java [hunk 4]
```diff
@@ -15,24 +19,80 @@
...
+			StudentRepository.save(student);
+			LOG.info("Étudiant avec ID {} mis à jour avec succès !", student.getId());
+		});
+
+		// Suppression d'un étudiant par ID
+		String StudentIdToDelete = "64bfca886bea761d7a15ca3f";
+		StudentRepository.deleteById(StudentIdToDelete);
+		LOG.info("Étudiant avec ID {} supprimé avec succès !", StudentIdToDelete);
+
+
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/BackendApplication.java:96]: Move that to a test in a class called StudentRepositoryTest (in reality we don't need this code, it's was just a way to lead how to interact with repositories and MongoDB with Spring/Java)

File: src/main/java/com/tajweed/backend/controller/AudioController.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+@RestController
+public class AudioController {
+
+    public static Logger logger = LoggerFactory.getLogger(AudioController.class);
+
+    @Autowired
+    private AudioRepository repository;
+
+
+    @RequestMapping("/students")
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/controller/AudioController.java:20]: wrong mapping, should be `/audios` ?

File: src/main/java/com/tajweed/backend/dao/StudentRepository.java [hunk 6]
```diff
@@ -4,4 +4,9 @@
 import org.springframework.data.repository.CrudRepository;
 
 public interface StudentRepository extends CrudRepository<Student, String> {
+<<<<<<< Updated upstream
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/dao/StudentRepository.java:7]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/dao/StudentRepository.java [hunk 7]
```diff
@@ -4,4 +4,9 @@
...
+=======
+    // Mise à jour d'un étudiant par ID
+    Student save(Student student); // Lorsque l'étudiant a un ID existant, cela mettra à jour l'enregistrement existant
+>>>>>>> Stashed changes
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/dao/StudentRepository.java:11]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/model/Student.java [hunk 8]
```diff
@@ -9,15 +10,28 @@ public class Student {
...
-    private String surname;
+    private String progress;
+
+    private String password;
+
+    private String email;
 
     private Integer age;
 
+<<<<<<< Updated upstream
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/model/Student.java:21]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/model/Student.java [hunk 9]
```diff
@@ -9,15 +10,28 @@ public class Student {
...
     public Student() {}
 
     public Student(String id, String name, String surname, Integer age){
+=======
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/model/Student.java:25]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/model/Student.java [hunk 10]
```diff
@@ -9,15 +10,28 @@ public class Student {
...
+    }
+
+
+    public Student(String id, String progress, String name, String password,String email){
+>>>>>>> Stashed changes
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/model/Student.java:30]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/model/Student.java [hunk 11]
```diff
@@ -9,15 +10,28 @@ public class Student {
...
         this.id = id;
+        this.progress = progress;
         this.name = name;
+<<<<<<< Updated upstream
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/model/Student.java:34]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/model/Student.java [hunk 12]
```diff
@@ -29,6 +43,10 @@ public String getName() {
     public void setName(String name) {
         assert name.isEmpty();
         this.name = name;
+=======
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/model/Student.java:46]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/model/Student.java [hunk 13]
```diff
@@ -29,6 +43,10 @@ public String getName() {
...
+        this.password = password;
+        this.email = email;
+>>>>>>> Stashed changes
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/model/Student.java:49]: should be removed (conflict not resolved ?)

File: src/test/java/com/tajweed/backend/AudioTest.java [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+    private MockMvc mockMvc;
+
+    @MockBean
+    private AudioRepository repository;
+
+
+    @Test
+    @WithMockUser(value = "user")
+    public void givenConnectedThenShouldReturnAllAudio() throws Exception {
+        when(repository.findAll()).thenReturn(List.of(new Audio("A", "duration", "idstudent", "idEntity" , "chapter", "verse")));
```
Comment (fodil-a) [inline src/test/java/com/tajweed/backend/AudioTest.java:33]: you should put realistic data (duration should be a duration, give a real chapter, a real verse, ...)

File: src/main/java/com/tajweed/backend/model/Audio.java [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+    private String id;
+
+    private String duration;
+    private String idStudent;
+
+    private String idEntity;
+
+    private String chapter;
+
+    private String verse;
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/model/Audio.java:17]: I believe verse should be a range ? (what if a student does Al fatiha 1 -> 7 ?)

File: src/test/java/com/tajweed/backend/StudentTest.java [hunk 16]
```diff
@@ -30,16 +30,28 @@ public class StudentTest {
     @Test
     @WithMockUser(value = "user")
     public void givenConnectedThenShouldReturnAllStudents() throws Exception {
+<<<<<<< Updated upstream
```
Comment (fodil-a) [inline src/test/java/com/tajweed/backend/StudentTest.java:33]: should be removed (conflict not resolved ?)

File: src/test/java/com/tajweed/backend/StudentTest.java [hunk 17]
```diff
@@ -30,16 +30,28 @@ public class StudentTest {
...
         when(repository.findAll()).thenReturn(List.of(new Student("A", "Fodil", "A", 27)));
         this.mockMvc.perform(get("/students"))
                 .andDo(print())
                 .andExpect(status().isOk())
                 .andExpect(content().json("[{'id': 'A', 'name': 'Fodil', 'surname': 'A', 'age': 27}]", true));
+=======
```
Comment (fodil-a) [inline src/test/java/com/tajweed/backend/StudentTest.java:39]: should be removed (conflict not resolved ?)

File: src/test/java/com/tajweed/backend/StudentTest.java [hunk 18]
```diff
@@ -30,16 +30,28 @@ public class StudentTest {
...
+        when(repository.findAll()).thenReturn(List.of(new Student("A", "En cours", "Fodil", "A" , "A")));
+        this.mockMvc.perform(get("/students"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(content().json("[{'id': 'A', 'progress': 'En cours', 'name': 'Fodil', 'password': 'A' , 'email': 'A'}]"));
+>>>>>>> Stashed changes
```
Comment (fodil-a) [inline src/test/java/com/tajweed/backend/StudentTest.java:45]: should be removed (conflict not resolved ?)

File: src/test/java/com/tajweed/backend/StudentTest.java [hunk 19]
```diff
@@ -30,16 +30,28 @@ public class StudentTest {
...
     }
 
     @Test
     public void givenNotConnectedThenShouldReturn401() throws Exception {
+<<<<<<< Updated upstream
```
Comment (fodil-a) [inline src/test/java/com/tajweed/backend/StudentTest.java:50]: should be removed (conflict not resolved ?)

File: src/test/java/com/tajweed/backend/StudentTest.java [hunk 20]
```diff
@@ -30,16 +30,28 @@ public class StudentTest {
...
         when(repository.findAll()).thenReturn(List.of(new Student("A", "Fodil", "A", 27)));
+=======
```
Comment (fodil-a) [inline src/test/java/com/tajweed/backend/StudentTest.java:52]: should be removed (conflict not resolved ?)

File: src/test/java/com/tajweed/backend/StudentTest.java [hunk 21]
```diff
@@ -30,16 +30,28 @@ public class StudentTest {
...
+        when(repository.findAll()).thenReturn(List.of(new Student("A", "En cours", "Fodil", "A" , "A")));
+>>>>>>> Stashed changes
```
Comment (fodil-a) [inline src/test/java/com/tajweed/backend/StudentTest.java:54]: should be removed (conflict not resolved ?)

**Review by fodil-a — COMMENTED** [review]
More on the review once fixed ! Good work, thanks 😊

File: src/main/java/com/tajweed/backend/BackendApplication.java [hunk 22]
```diff
@@ -15,24 +19,80 @@
...
 	}
 
+<<<<<<< Updated upstream
 	private static Logger LOG = LoggerFactory
 			.getLogger(BackendApplication.class);
 
 	@Autowired
 	private StudentRepository studentRepository;
 
+=======
```
Comment (fodil-a) [inline src/main/java/com/tajweed/backend/BackendApplication.java:60]: should be removed (conflict not resolved ?)

File: src/main/java/com/tajweed/backend/model/Audio.java (hunk 15 above)
Comment (Pseudonyme7) [reply src/main/java/com/tajweed/backend/model/Audio.java:16]: that's right it has to be a range, verses were previously used for demo purposes

**Review by Pseudonyme7 — APPROVED** [review]

**Pull request merged**